lib/modules/manager/dockerfile/extract.ts

```diff
@@ -88,6 +88,7 @@ function processDepForAutoReplace(
   dep.replaceString = unfoldedLineNumbers
     .map((lineNumber) => lines[lineNumber])
     .join(linefeed);
+  dep.replaceString += linefeed;
```

Comment:
Won't this break if the line somehow had a character before the line feed, such as whitespace? e.g. could the line be `ARG abc=1.0.0 `? Therefore I'd append the "next" character in the line and if none *then* newline. What do you think?

Reply:
i think it works as expected now.

@Churro please add or change a test to validate trailing content.

Reply:
Thank you for bringing up the `ARG abc=1.0.0 ` case. It revealed a deficiency that was addressed in PR https://github.com/renovatebot/renovate/pull/18592.

Adding a linefeed here is a "general" solution in the sense that a linefeed is always the next character to come. This is because `dep.replaceString` in line 88 comprises the entire line, incl potentially trailing spaces. 

There is a test now that exemplifies how this works with trailing spaces:
https://github.com/renovatebot/renovate/blob/056ff8c0fc61492ca538ca6ef938847fa7ea8854/lib/modules/manager/dockerfile/extract.spec.ts#L764-L782